contrib/install.sh

```diff
@@ -0,0 +1,321 @@
+  sudo tee $florestaLib/config.toml > /dev/null <<EOF
+[wallet]
+xpubs = [$([ -n "$wallet_xpub" ] && echo "$wallet_xpub")]
+descriptors = [$([ -n "$wallet_descriptor" ] && echo "$wallet_descriptor")]
+addresses = []
+EOF
+
+  # Build service file
+  echo "🐧 Generating $florestaService (need sudo)"
+  sudo tee $florestaService > /dev/null <<EOF
+[Unit]
+Description=Floresta: A Lightweight Utreexo-powered Bitcoin full node implementation
+Documentation=https://github.com/vinteumorg/Floresta
+After=network-online.target time-set.target
+Wants=network-online.target
+
+[Service]
+ExecStart=/usr/local/bin/florestad --daemon --network=$network --data-dir=$florestaDir --config-file=$florestaLib/config.toml --pid-file=$florestaRun/florestad.pid --log-to-file $([ "$assume_utreexo" = true ] && echo "--assume-utreexo")$([ "$enable_ssl" = true ] && echo "--ssl-cert-path /etc/florestad/ssl/florestad.crt --ssl-key-path /etc/florestad/ssl/florestad.key --ssl-electrum-address 0.0.0.0:50002")
+
+# Ensure that the service is ready after the MainPID exists
+Type=forking
+PIDFile=$florestaRun/florestad.pid
+
+# Don't enter a restart loop, as it might corrupt our database
+Restart=no
+
+TimeoutStartSec=infinity
+TimeoutStopSec=600
+
+# Make sure we can read from the config file
+ExecStartPre=/bin/chgrp florestad /etc/florestad
+User=florestad
+Group=florestad
+
+# /run/florestad
+RuntimeDirectory=florestad
+RuntimeDirectoryMode=0710
+
+# /etc/florestad
+ConfigurationDirectory=florestad
+ConfigurationDirectoryMode=0710
+
+# /var/lib/florestad
+StateDirectory=florestad
+StateDirectoryMode=0710
+
+# Provide a private /tmp and /var/tmp.
+PrivateTmp=true
+
+# Mount /usr, /boot/ and /etc read-only for the process.
+ProtectSystem=full
+
+# Deny access to /home, /root and /run/user
+ProtectHome=true
+
+# Disallow the process and all of its children to gain
+# new privileges through execve().
+NoNewPrivileges=true
+
+# Use a new /dev namespace only populated with API pseudo devices
+# such as /dev/null, /dev/zero and /dev/random.
+PrivateDevices=true
+
+# Deny the creation of writable and executable memory mappings.
+MemoryDenyWriteExecute=true
+
+# Restrict ABIs to help ensure MemoryDenyWriteExecute is enforced
+SystemCallArchitectures=native
+
+[Install]
+WantedBy=multi-user.target
+EOF
+}
+
+# show some useful information before start floresta node
+show_done() {
+  echo "✔️ DONE"
+  echo ""
+  echo "Before enable/start, please edit '$florestaService' to your needs. After that, run:"
+  echo ""
+  echo "    sudo systemctl enable florestad.service # this enable service on boot" 
+  echo "    sudo systemctl start florestad.service  # this will start the service now"
+  echo "    sudo systemctl status florestad.service # this check if service is running well"
+  echo "    floresta_cli getblockchaininfo          # this assures all OK"
+  echo ""
+}
+
+# MAIN SECTION
+# Run all functions on sequence
+apt_install gcc build-essential pkg-config libssl-dev mold
+install_rustup
```

Comment:
I think we would need a clean-up here. Maybe our user doesn't want rust installed in their systems?